lms/static/js/discovery/main.js

```diff
@@ -0,0 +1,22 @@
+RequireJS.require([
+    'jquery',
+    'backbone',
+    'js/discovery/app',
+    'js/discovery/collection',
+    'js/discovery/form',
+    'js/discovery/result_list_view',
+    'js/discovery/filter_bar_view',
+    'js/discovery/search_facets_view'
+], function ($, Backbone, App, Collection, DiscoveryForm, ResultListView, FilterBarView, FacetsBarView) {
```

Comment:
I agree with Will's comments. Unfortunately RequireJS is still a work-in-progress in the LMS, and there are many dimensions that need to be fixed (usage of Optimizer, caching strategy, RequireJS text etc). I have a wiki page where I've been collecting all these issues:

https://openedx.atlassian.net/wiki/display/TNL/RequireJS+in+the+LMS

I also have another page that collects the current best practices, but it doesn't really help you here:

https://openedx.atlassian.net/wiki/display/TNL/How+to+add+a+new+page+to+LMS+or+Studio